third_party/move/move-compiler/transactional-tests/tests/control_flow/nested_for_while_loops.move

```diff
@@ -0,0 +1,20 @@
+//# run
+script {
+    fun main(): () {
+        let y = 0;
+        for (i in 0..10) {
+            y = y + 1;
+            while (y > 1) {
+                y = y - 1;
+            };
+        };
+        assert!(y == 1, 42);
```

Comment:
This is an odd test case.  No matter how many times the first loop is traversed, y==1 here.

Reply:
Updated the test.